File: online-judge-solutions/Codeforces/1671A.py
#https://codeforces.com/problemset/problem/1671/A
#Lang		:	Python 3.8
#Time		:	62 ms
#Memory         :	0 KB
for _ in range(int(input())):
    s=input()
    if(len(s)==1 or s.find("aba")!=-1 or s.find("bab")!=-1 or s[:2]=="ab" or s[:2]=="ba" or s[len(s)-2:]=="ab" or s[len(s)-2:]=="ba"): print("NO")
    else: print("YES")

File: online-judge-solutions/Codeforces/1671A.cpp
//https://codeforces.com/problemset/problem/1671/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int t;
	cin>>t;
	while(t--){
		string s;cin>>s;
		if(s.size()==1 || (s.find("aba")<s.size()) || (s.find("bab")<s.size()) || s.substr(0,2)=="ab" || s.substr(0,2)=="ba" || s.substr(s.size()-2,2)=="ab" || s.substr(s.size()-2,2)=="ba") cout<<"NO\n";
		else cout<<"YES\n";
	}
	return 0;
}
